scripts/generate_index_pages.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * Generates index.html pages for the home page and all state / county pages,
+ * filling the meta tags for OpenGraph / Twitter with the appropriate title /
+ * description / image information.
+ *
+ * You can run via `yarn generate-index-pages`
+ */
+
+import fs from 'fs-extra';
+import path from 'path';
+import _ from 'lodash';
+import urlJoin from 'url-join';
+import US_STATE_DATASET from '../src/components/MapSelectors/datasets/us_states_dataset_01_02_2020.json';
+import ShareImageUrlJSON from '../src/assets/data/share_images_url.json';
+import { STATES } from '../src/common';
+
+// We don't care about the values here, but this is a cheap way to determine all
+// of the counties we have any data for and are therefore share-able.
+import CalculatedCountyInterventionJSON from '../src/assets/data/calculated_county_interventions.json';
+const COUNTIES = Object.keys(CalculatedCountyInterventionJSON);
+
+type MetaTags = {[name: string]: string}
+
+function homePageTags(imageUrl: string): MetaTags {
+  // Most of the tags are set correctly in public/index.html so we don't need to
+  // replace them.
+  return {
+    'og:image:url': imageUrl,
+    'twitter:image': imageUrl,
+  };
+}
+
+function locationPageTags(fullImageUrl: string, canonicalUrl: string, locationName: string): MetaTags {
+  const title = `Is ${locationName} ready to reopen?`;
+  const description = 'Real-time modeling and metrics to understand where we stand against COVID.';
+  return {
+    'og:url': canonicalUrl,
+    'og:image:url': fullImageUrl,
+    'og:title': title,
+    'og:description': description,
+
+    'twitter:image': fullImageUrl,
+    'twitter:title': title,
+    'twitter:description': description,
+  };
+}
+
+async function main() {
+  console.log('Building index.html pages...');
+
+  const builder = await IndexPageBuilder.initialize();
+
+  await builder.writeTemplatedPage('/index.html', homePageTags(builder.fullImageUrl('home.png')));
+
+  for (const stateCode in STATES) {
+    const stateName = (STATES as any)[stateCode];
+    const relativeUrl = `/us/${stateCode.toLowerCase()}/`;
+    const canonicalUrl = urlJoin('https://covidactnow.org/', relativeUrl);
+    const imageUrl = builder.fullImageUrl(`states/${stateCode.toLowerCase()}.png`);
+    const page = path.join(relativeUrl, 'index.html');
+    await builder.writeTemplatedPage(page, locationPageTags(imageUrl, canonicalUrl, stateName));
+  }
+
+  for (const fips of COUNTIES) {
+    const county = findCountyByFips(fips);
+    if (!county) {
+      console.warn('Skipping missing county: ', fips);
+      continue;
```

Comment:
Just an FYI, it seems that there is only one missing

```
Building index.html pages...
Skipping missing county:  11001
```

Reply:
Oh, yeah...  That's Washington, DC which we treat as a state, not a county.   For the other PR, I explicitly blacklisted it, so I went ahead and did the same thing here.